packages/gatsby/cache-dir/socketIo.js

```diff
@@ -11,6 +11,10 @@ let pageQueryData = {}
 export const getStaticQueryData = () => staticQueryData
 export const getPageQueryData = () => pageQueryData
 
+const didDataChange = (id, queryData, exisitingQueryData) =>
+  !(id in exisitingQueryData) ||
+  JSON.stringify(exisitingQueryData) !== JSON.stringify(queryData[id])
```

Comment:
I think this should be other way around (`exisitingQueryData` is the id -> cached results and `queryData` is "new" result:
```suggestion
  JSON.stringify(queryData) !== JSON.stringify(exisitingQueryData[id])
```